src/rules/selector-max-specificity/__tests__/index.js

```diff
@@ -0,0 +1,35 @@
+import {
+  ruleTester,
+  warningFreeBasics,
+} from "../../../testUtils"
+import rule, { ruleName, messages } from ".."
+
+const testRule = ruleTester(rule, ruleName)
+
+testRule(30, tr => {
+  warningFreeBasics(tr)
+
+  tr.ok(".ab {}")
+  tr.ok(".ab .cd {}")
+  tr.ok(".ab .cd span {}")
+  tr.ok(".cd div span {}")
+  tr.ok(".cd .de div span a {}")
+  tr.ok(".cd .de div span a > b {}")
+  tr.ok(".cd .de, .cd .ef > b {}")
+
+  tr.notOk("#jubjub {}", {
+    message: messages.expected("#jubjub", 30),
+    line: 1,
+    column: 1,
+  })
+  tr.notOk(".thing div .thing .sausages {}", {
+    message: messages.expected(".thing div .thing .sausages", 30),
+    line: 1,
+    column: 1,
+  })
+  tr.notOk(".thing div .thing, .sausages .burgers .bacon a {}", {
+    message: messages.expected(".sausages .burgers .bacon a", 30),
+    line: 1,
+    column: 20,
+  })
+})
```

Comment:
You might want to [install a `editorconfig` plugin](http://editorconfig.org/#download) for your code editor. It will automatically configure your editor for consistent whitespace e.g. adding an empty line at the bottom of all the files.